android/src/main/java/vn/hunghd/flutterdownloader/IntentUtils.java

```diff
@@ -38,9 +38,10 @@ public static synchronized Intent validatedFileIntent(Context context, String pa
         String mime = null;
         try {
             FileInputStream inputFile = new FileInputStream(path);
-            mime = URLConnection.guessContentTypeFromStream(inputFile);
+            mime = URLConnection.guessContentTypeFromStream(inputFile);//does not work in some target sdk version
             if(mime==null){
-                mime = URLConnection.guessContentTypeFromStream(new BufferedInputStream(inputFile));
+                mime = URLConnection.guessContentTypeFromName(path);//works fine
```

Comment:
if `URLConnection.guessContentTypeFromStream` thows an exception, the below codes will be never executed